rclcpp/include/rclcpp/node.hpp

```diff
@@ -135,7 +135,53 @@ class Node
   create_subscription(
     const std::string & topic_name,
     const rmw_qos_profile_t & qos_profile,
-    std::function<void(const std::shared_ptr<MessageT> &)> callback,
+    typename rclcpp::subscription::AnySubscriptionCallback<MessageT>::SharedPtrCallback callback,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr,
+    bool ignore_local_publications = false,
+    typename rclcpp::message_memory_strategy::MessageMemoryStrategy<MessageT>::SharedPtr
+    msg_mem_strat = nullptr);
+
+  template<typename MessageT>
+  typename rclcpp::subscription::Subscription<MessageT>::SharedPtr
+  create_subscription(
+    const std::string & topic_name,
+    const rmw_qos_profile_t & qos_profile,
+    typename rclcpp::subscription::AnySubscriptionCallback<MessageT>::SharedPtrWithInfoCallback
+    callback,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr,
+    bool ignore_local_publications = false,
+    typename rclcpp::message_memory_strategy::MessageMemoryStrategy<MessageT>::SharedPtr
+    msg_mem_strat = nullptr);
+
+  template<typename MessageT>
+  typename rclcpp::subscription::Subscription<MessageT>::SharedPtr
+  create_subscription(
+    const std::string & topic_name,
+    const rmw_qos_profile_t & qos_profile,
+    typename rclcpp::subscription::AnySubscriptionCallback<MessageT>::UniquePtrCallback callback,
+    rclcpp::callback_group::CallbackGroup::SharedPtr group = nullptr,
+    bool ignore_local_publications = false,
+    typename rclcpp::message_memory_strategy::MessageMemoryStrategy<MessageT>::SharedPtr
+    msg_mem_strat = nullptr);
+
+  template<typename MessageT>
+  typename rclcpp::subscription::Subscription<MessageT>::SharedPtr
+  create_subscription(
+    const std::string & topic_name,
+    const rmw_qos_profile_t & qos_profile,
+    typename rclcpp::subscription::AnySubscriptionCallback<MessageT>::UniquePtrWithInfoCallback
```

Comment:
Actually, I don't think it's possible since we need to partially specialize them and that doesn't seem to be allowed for function templates and/or member function templates.